lib/cheffish/base_resource.rb

```diff
@@ -1,7 +1,20 @@
 require 'chef_compat/resource'
+require 'cheffish/array_property'
 
 module Cheffish
   class BaseResource < ChefCompat::Resource
+    def initialize(*args)
+      super
+      chef_server run_context.cheffish.current_chef_server
```

Comment:
Nope, we still need to do this because we need the value of current chef server *at the time of initialization* (lazy defaults won't get evaluated until later). Otherwise, someone could create the resource and then later change with_chef_server, and we'd pick up the new value when we finally lazy evaluated `chef_server`.